### PR TITLE
#[187364670] - Chore(Jest Testing): Adding a testing mechanism with Jest library

### PR DESCRIPTION
**Pull request by AngeloChristian1**
### Summary
This PR introduces Jest testing to our e-commerce website frontend project. The primary motivation behind this change is to enhance code quality and ensure that our application functions correctly across different scenarios.

### Testing Approach
I have implemented unit tests using Jest to cover critical components, functions, and user interactions. These tests validate the correctness of our code and help prevent regressions.
To test simply type `npm run test` or `npm t`
The test coverage is set to 80 but it can be changed further

### Checklist
-  Code adheres to project style guidelines
-  Self-reviewed my changes
-  Added comments to clarify complex sections
-  No new warnings introduced
-  Created tests to verify the effectiveness of my changes
-  Verified that existing unit tests pass locally

### Relevant Stories
This PR addresses the following pivotal tracker/Trello story:
[#Delivers 187364670]

### Screenshots (if appropriate)
An example of tests that are running (loom video)
https://www.loom.com/share/7bdb1cf1ea8c482dbee5a39911170d37?sid=7713f6c2-d270-4525-ad77-eb18f147e2b6

### CODE REVIEW

**Review by Yvan-David — APPROVED** [review]
Great Job

File: src/test/App.test.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  it('should handle increment', () => {
+    const result = counterReducer(initialState, increment());
+    expect(result.count).toEqual(1);
+  });
+
+  it('should handle decrement', () => {
+    const result = counterReducer(initialState, decrement());
+    expect(result.count).toEqual(-1);
+  });
```
Comment (mpanoathos) [inline src/test/App.test.tsx:45]: Nice

**Review by mpanoathos — APPROVED** [review]

File: jest.config.cjs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/** @type {import('ts-jest').JestConfigWithTsJest} */
+
+  module.exports = {
+    preset: 'ts-jest',
+    testEnvironment: 'jest-environment-jsdom',
+    collectCoverage: true,
+    coverageDirectory: 'coverage',
+    coverageReporters: ['json', 'lcov', 'text', 'clover'],
+    coverageThreshold: {
```
Comment (munezerobagira) [inline jest.config.cjs:13]:
```suggestion
  collectCoverageFrom: [
     '**/*.{js,jsx}',
     '!**/node_modules/**',
    ],
    coverageThreshold: {
```

Comment (munezerobagira) [reply jest.config.cjs:13]: We need to make sure that the coverage is collected from all files

**Review by munezerobagira — CHANGES_REQUESTED** [review]
Nice work but there few things which can improve

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**